src/Cms/Blueprint.php

```diff
@@ -607,11 +611,16 @@ public static function fieldsProps($fields): array
 
 			// resolve field groups
 			if ($fieldProps['type'] === 'group') {
-				if (empty($fieldProps['fields']) === false && is_array($fieldProps['fields']) === true) {
+				if (
+					empty($fieldProps['fields']) === false &&
+					is_array($fieldProps['fields']) === true
+				) {
 					$index  = array_search($fieldName, array_keys($fields));
-					$before = array_slice($fields, 0, $index);
-					$after  = array_slice($fields, $index + 1);
-					$fields = array_merge($before, $fieldProps['fields'] ?? [], $after);
+					$fields = [
+						...array_slice($fields, 0, $index),
+						...$fieldProps['fields'] ?? [],
+						...array_slice($fields, $index + 1)
+					];
```

Comment:
This use is great though as it makes the merge order much easier to grasp than the "before, after, middle" order that was there before.

Reply:
Yes, I think the syntax is easier to understand and grasp overall as it puts the equally important parts at the same level side by side vertically here.

```php
$array = [
   ...$otherArray,
   'foo' => 'bar'
];
```

is in my view much quicker to grasp as

```php
$array = array_merge($otherArray, [
   'foo' => 'bar'
];
```

where one part is hiding in the same line of the function call, while other parts are spread out vertically.